csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -237,7 +329,9 @@ public ReadOnlySpan<byte> GetBytes(int index)
 
             if (IsNull(index))
             {
-                return null;
+                // Note that `return null;` is valid syntax, but would be misleading as `null` in the context of a span
+                // is actually returned as an empty span.
+                return ReadOnlySpan<byte>.Empty;
```

Comment:
This clause previously said `return null;`, which was misleading as it suggested that the method could be used to identify null values in the array.  However, `null` in the context of a `ReadOnlySpan<T>` is the same as the empty span, and so the method can't distinguish null values from empty ones:

```csharp
Assert.True(ReadOnlySpan<byte>.Empty == null)
```

The new code is clearer in intent.

Reply:
I've been confused and bit by this a few times in different projects. All the following ways produce an "equivalent" span:

* `return null`  (this uses the implicit operator from an array)
* `return default`
* `return ReadOnlySpan<T>.Empty`  (this just says `return default` underneath the covers)

I agree that `ReadOnlySpan<T>.Empty` is the most clear thing to return here. Next would be `return default`. `return null` is the worst of the three options IMO.